examples/detection/yolov2/yolov2/links/yolov2.py

```diff
@@ -0,0 +1,328 @@
+# Mofidied by:
+# Copyright (c) 2017 Yuki Furuta
+#
+# Original work by:
+# --------------------------------------------------------
+# YOLOv2
+# Copyright (c) 2017 leetenki
+# Licensed under The MIT License [see LICENSE for details]
+# https://github.com/leetenki/YOLOv2
+# --------------------------------------------------------
+
+from __future__ import print_function
+from __future__ import absolute_import
+
+import os
+import numpy as np
+from chainer.dataset import download
+from chainer.serializers import npz
+from chainer import Variable
+from chainer import Chain
+import chainer.links as L
+import chainer.functions as F
+from yolov2.functions import reorg
+from yolov2.utils import convert_darknet_model_to_npz
+
+
+class YOLOv2(Chain):
+
+    """
+    YOLOv2
+    - It takes (416, 416, 3) sized image as input
+    """
+
+    def __init__(self, n_classes, n_boxes, pretrained_model='auto'):
+        super(YOLOv2, self).__init__(
+            ##### common layers for both pretrained layers and yolov2 #####
+            conv1  = L.Convolution2D(3, 32, ksize=3, stride=1, pad=1, nobias=True),
```

Comment:
> Could you follow this instruction and fix style errors?
>
> ```
> pip install flake8
> pip install hacking
> cd examples
> flake8 .
> ```

```bash
./detection/yolov2/yolov2/links/yolov2.py:37:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:37:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:37:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:37:80: E501 line too long (83 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:38:16: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:38:16: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:38:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:39:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:39:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:39:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:40:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:40:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:40:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:40:80: E501 line too long (84 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:41:16: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:41:16: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:41:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:42:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:42:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:42:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:43:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:43:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:43:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:43:80: E501 line too long (85 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:44:16: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:44:16: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:44:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:45:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:45:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:45:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:46:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:46:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:46:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:46:80: E501 line too long (85 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:47:16: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:47:16: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:47:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:48:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:48:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:48:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:49:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:49:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:49:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:49:80: E501 line too long (85 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:50:16: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:50:16: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:50:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:51:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:51:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:51:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:52:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:52:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:52:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:52:80: E501 line too long (86 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:53:16: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:53:16: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:53:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:54:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:54:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:54:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:55:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:55:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:55:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:55:80: E501 line too long (86 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:56:16: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:56:16: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:56:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:57:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:57:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:57:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:58:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:58:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:58:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:58:80: E501 line too long (86 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:59:16: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:59:16: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:59:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:60:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:60:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:60:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:61:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:61:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:61:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:61:80: E501 line too long (86 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:62:16: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:62:16: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:62:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:63:18: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:63:18: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:63:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:64:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:64:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:64:80: E501 line too long (86 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:65:17: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:65:17: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:65:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:66:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:66:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:67:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:67:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:67:80: E501 line too long (86 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:68:17: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:68:17: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:68:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:69:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:69:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:70:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:70:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:70:80: E501 line too long (86 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:71:17: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:71:17: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:71:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:72:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:72:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:73:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:73:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:73:80: E501 line too long (86 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:74:17: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:74:17: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:74:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:75:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:75:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:76:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:76:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:76:80: E501 line too long (87 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:77:17: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:77:17: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:77:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:78:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:78:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:79:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:79:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:79:80: E501 line too long (87 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:80:17: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:80:17: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:80:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:81:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:81:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:82:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:82:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:82:80: E501 line too long (87 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:83:17: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:83:17: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:83:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:84:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:84:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:85:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:85:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:85:80: E501 line too long (87 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:86:17: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:86:17: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:86:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:87:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:87:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:88:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:88:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:88:80: E501 line too long (87 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:89:17: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:89:17: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:89:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:90:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:90:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:92:13: E266 too many leading '#' for block comment
./detection/yolov2/yolov2/links/yolov2.py:93:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:93:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:93:80: E501 line too long (88 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:94:17: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:94:17: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:94:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:95:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:95:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:96:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:96:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:96:80: E501 line too long (88 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:97:17: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:97:17: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:97:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:98:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:98:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:99:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:99:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:99:80: E501 line too long (88 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:100:17: E221 multiple spaces before operator
./detection/yolov2/yolov2/links/yolov2.py:100:17: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:100:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:101:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:101:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:102:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:102:21: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:102:80: E501 line too long (109 > 79 characters)
./detection/yolov2/yolov2/links/yolov2.py:103:19: E251 unexpected spaces around keyword / parameter equals
./detection/yolov2/yolov2/links/yolov2.py:103:21: E251 unexpected spaces around keyword / parameter equal
```

If I follow the instruction by `flake8`, I think, this part will be very dirty. Should I follow the instruction anyway?